config/genesis.go

```diff
@@ -0,0 +1,82 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
+	apiConfig "github.com/spacemeshos/go-spacemesh/api/config"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/hash"
+	"github.com/spacemeshos/go-spacemesh/log"
+)
+
+const (
+	DefaultGenesisConfigFileDir  = "data"
+	DefaultGenesisConfigFileName = "genesis.conf"
+	defaultGenesisExtraData      = "mainnet"
+
+	DefaultTestGenesisTime      = "2022-12-25T00:00:00+00:00"
+	defaultTestGenesisExtraData = "test"
+)
+
+type GenesisConfig struct {
+	Accounts    map[string]uint64 `mapstructure:"accounts"`
+	GenesisTime string            `mapstructure:"genesis-time"`
+	ExtraData   string            `mapstructure:"genesis-extradata"`
+}
+
+func defaultGenesisConfig() *GenesisConfig {
+	return &GenesisConfig{
+		Accounts:    apiConfig.DefaultGenesisAccounts(),
+		GenesisTime: time.Now().Format(time.RFC3339),
+		ExtraData:   defaultGenesisExtraData,
+	}
+}
+
+func GenesisDataDir(dataDirParent string) string {
```

Comment:
can you explain why do we need this?

Reply:
Removed along with https://github.com/spacemeshos/go-spacemesh/pull/3568#discussion_r977193346